### PR TITLE
i#4180 no-alloc IR: Add documentation

### PR DESCRIPTION
**Pull request by derekbruening**
Adds a section to the IR docs on the new no-alloc IR feature.

Issue: #4180

### CODE REVIEW

File: api/docs/bt.dox [hunk 1]
```diff
@@ -747,6 +751,30 @@ and explicit).  The sources are listed first, followed by "->", and then
 the destinations.  This provides more information than any of the other
 formats.
 
+********************
+\subsection sec_IR_heap Instruction Heap Allocation
+
+DynamoRIO's IR is designed for efficiency and a small footprint.  By
+default, space for operands is dynamically allocated from the heap.
+This can be problematic when using instructions in fragile locations
+such as signal handlers.  DR provides a separate insruction structure
```
Comment (johnfxgalea) [inline api/docs/bt.dox]: typo: instruction

File: api/docs/bt.dox [hunk 2]
```diff
@@ -747,6 +751,30 @@ and explicit).  The sources are listed first, followed by "->", and then
...
+for such situations: #instr_noalloc_t.  This structure contains
+built-in storage for all possible operand slots and for temporary
+encoding space, avoiding heap allocation when used for decoding or
+encoding.
+
+To use #instr_noalloc_t, declare it and then obtain a pointer to it as an #instr_t structure for use with API functions:
```
Comment (johnfxgalea) [inline api/docs/bt.dox]: Add a new line?

File: api/docs/bt.dox (hunk 1 above)
Comment (derekbruening) [reply api/docs/bt.dox]: ty, fixed.

File: api/docs/bt.dox (hunk 2 above)
Comment (derekbruening) [reply api/docs/bt.dox]: Done.

**Comment by derekbruening** [comment]
A64 (should be fixed in next few days): #4189.  Appveyor: #4058.

**Pull request merged**